marathon-bigip-ctlr.py

```diff
@@ -761,36 +925,42 @@ def do_reset(self):
 
                     self.__apps = get_apps(self.__marathon.list(),
                                            self.__marathon.health_check())
-                    cfg = create_config_marathon(self.__bigip, self.__apps)
-                    if self.__bigip.regenerate_config_f5(cfg):
-                        # Timeout (or some other retryable error occurred),
-                        # do a reset so that we try again
-                        self.retry_backoff(self.reset_from_tasks)
-                    else:
-                        # Reconfig was successful
-                        self.start_checkpoint_timer()
-                        self._backoff_timer = 1
-
-                        perf_enable = os.environ.get('SCALE_PERF_ENABLE')
-                        if perf_enable:  # pragma: no cover
-                            test_data = {}
-                            app_count = 0
-                            backend_count = 0
-                            for app in self.__apps:
-                                if app.partition == 'test':
-                                    app_count += 1
-                                    backends = len(app.backends)
-                                    test_data[app.appId[1:]] = backends
-                                    backend_count += backends
-                            test_data['Total_Services'] = app_count
-                            test_data['Total_Backends'] = backend_count
-                            test_data['Time'] = time.time()
-                            json_data = json.dumps(test_data)
-                            logger.info('SCALE_PERF: Test data: %s', json_data)
-
-                    logger.debug("updating tasks finished, took %s seconds",
-                                 time.time() - start_time)
-
+                    for cccl in self.__cccls:
+                        cfg = create_config_marathon(cccl, self.__apps)
```

Comment:
Gary's comment that misconfig of one partition shouldn't prevent us from configing other partitions.

Sounds like we agreed on refactoring this a bit to have per-CCCL exception handling, and then handling backoff once-per-pass, where "pass" is "update all CCCLs once"